nxdrive/gui/application.py

```diff
@@ -839,7 +839,11 @@ def confirm_cancel_transfer(
     def open_authentication_dialog(
         self, url: str, callback_params: Dict[str, str]
     ) -> None:
+        from PyQt5.QtCore import Qt
+        from PyQt5.QtWidgets import QApplication
+
         self.api.callback_params = callback_params
+        QApplication.setOverrideCursor(Qt.WaitCursor)
```

Comment:
- Only change the cursor when the app is frozen.
- Use a try/finally block to ensure the cursor is always restored.